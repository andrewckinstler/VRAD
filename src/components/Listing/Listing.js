import React from 'react'
import './Listing.scss';

const Listing = (props) => {
  return (
    <article>
      <h2>{props.listing.name}</h2>
      <h3>{props.listing.address.street}</h3>
    </article>
  )
}


export default Listing;
