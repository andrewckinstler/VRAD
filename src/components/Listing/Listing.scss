@mixin borders() {
  border: 1px solid #EBEBEB;
  border-radius: 4px;
  box-shadow: 0 2px 4px rgba(0,0,0,0.1);
}

@mixin buttons() {
  border-radius: .2em;
  color: #0C4C38;
  font-size: .8em;
  font-weight: bold;
  height: 2.5em;
  width: 20%;

  &:hover {
    cursor: pointer;
    background-color: #28BB8D;
  }
}

article {
  @include borders();
  margin-bottom: 2em;
  padding: .8em;
}

.preview-img {
  border-radius: .3em;
  height: 15em;
  object-fit: cover;
  width: 100%;
}

.details-btn {
  @include borders();
  @include buttons();
}

.listing-preview {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: .8em;
}

.listing-name {
  margin: 0;
}
