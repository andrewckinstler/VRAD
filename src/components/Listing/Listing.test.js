import React from 'react';
import { shallow } from 'enzyme';
import Listing from './Listing';

describe('Listing', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    let mockListing = {
      listing_id: 5,
      name: 'test name',
      key: 'test key'
    }
    const wrapper = shallow(<Listing
      key={mockListing.key}
      listing={mockListing}
    />);
  expect(wrapper).toMatchSnapshot();
  });
});
