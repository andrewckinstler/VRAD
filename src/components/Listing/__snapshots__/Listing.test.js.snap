// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing should match the snapshot with all data passed in correctly 1`] = `
<article>
  <img
    alt="Listing"
    className="preview-img"
    src="/images/5_a.jpg"
  />
  <div
    className="listing-preview"
  >
    <h2
      className="listing-name"
    >
      test name
    </h2>
    <button
      className="details-btn"
    >
      Details
    </button>
  </div>
</article>
`;
